core/src/main/scala/chisel3/Mem.scala

```diff
@@ -45,7 +45,7 @@ object Mem {
     do_apply(BigInt(size), t)(sourceInfo, compileOptions)
 }
 
-sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends HasId with NamedComponent with SourceInfoDoc {
+sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt, clockInst: Clock = Builder.forcedClock) extends HasId with NamedComponent with SourceInfoDoc {
   _parent.foreach(_.addId(this))
 
```

Comment:
```suggestion
sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends HasId with NamedComponent with SourceInfoDoc {
  _parent.foreach(_.addId(this))

  private val clockInst: Clock = Builder.forcedClock
```

This class is sealed and we never seem to actually pass a different clock as an argument so I think it's more clear just to make it a private val.